src/main/treeDataProviders/utils/analyzerUtils.ts

```diff
@@ -49,6 +51,41 @@ export class AnalyzerUtils {
         return fileNode;
     }
 
+    /**
+     * Transform the exclude pattern to patterns for the applicable scan.
+     * The following actions will be preformed to the pattern:
+     * 1. If validation for the exclude pattern fails no exclude will be returned.
+     * 2. If pattern contains {}, it will be splitted to multiple patterns, one for each option
+     * 3. '/**' will be add at the suffix to convert the pattern to match files and not folders
+     * @param excludePattern - the pattern to transform
+     * @param log - if log is passed the method will log the error that occur during conversion
+     * @returns the applicable pattern array
+     */
+    public static getApplicableExcludePattern(excludePattern?: string, log?: LogManager): string[] {
```

Comment:
This funtion is very busy doing and have a lot of different things. Instead, try to split it. The smaller the functions, the easier it is to read and understand them.